shared/utilities/getTicketStatus/test.js

```diff
@@ -0,0 +1,130 @@
+    tk.freeze(new Date(100));
+    const ticketReleaseDate = new Date(101);
+    const result = getTicketStatusOptions({ ticketReleaseDate });
+
+    expect(result).to.deep.equal({
+      buttonText: 'Free Ticket',
+      linkType: '',
+      statusHeader: 'TICKETS WILL GO LIVE ON',
+      statusSubHeader: 'Thursday, 1st January 1970, 01:00',
+      link: undefined,
+    });
+  });
+
+  it('returns Free ticket and Event linkType if the current Time is greater than the release date and there are tickets availiable', () => {
+    tk.freeze(new Date(100));
+    const ticketReleaseDate = new Date(99);
+    const result = getTicketStatusOptions({ ticketReleaseDate, ticketsAvailable: true, waitingListOpen: false });
+
+    expect(result).to.deep.equal({
+      buttonText: 'Free Ticket',
+      linkType: 'TICKET',
+      statusHeader: 'TICKETS LIVE',
+      statusSubHeader: 'Go to {Eventbrite} to get yours',
+      link: undefined,
+    });
+  });
+
+  it('returns Join Waitlist and Event linkType if the waiting list is open', () => {
+    tk.freeze(new Date(100));
+    const ticketReleaseDate = new Date(99);
+    const result = getTicketStatusOptions({ ticketReleaseDate, ticketsAvailable: false, waitingListOpen: true });
+
+    expect(result).to.deep.equal({
+      buttonText: 'Join Waitlist',
+      linkType: 'TICKET',
+      statusHeader: 'TICKETS NOW SOLD OUT',
+      statusSubHeader: 'Join the waiting list on {eventbrite}',
+      link: undefined,
+    });
+  });
+
+  it('returns Join Live Stream and Stream linkType if the current time is greater than the release date time and there are no tickets and no wait list', () => {
+    tk.freeze(new Date(100));
+    const result = getTicketStatusOptions({ ticketReleaseDate: new Date(99), endDateTime: new Date(101), ticketsAvailable: false, waitingListOpen: false });
+
+    expect(result).to.deep.equal({
+      buttonText: 'Join Live Stream',
+      linkType: 'STREAM',
+      statusHeader: 'TICKETS NOW SOLD OUT',
+      statusSubHeader: 'Couldn’t get a ticket? We got your back.',
+      link: undefined,
+    });
+  });
+
+  it('returns Watch and Stream linkType if the event is completed', () => {
+    tk.freeze(new Date(100));
+// <<<<<<< Updated upstream
+//     const result = getTicketStatusOptions({ endDateTime: new Date(99), ticketsAvailable: false, waitingListOpen: false });
+// =======
+    const result = getTicketStatusOptions({ ticketReleaseDate: new Date(98), endDateTime: new Date(99), ticketsAvailable: false, waitingListOpen: false });
+// >>>>>>> Stashed changes
+
+    expect(result).to.deep.equal({
+      buttonText: 'Watch',
+      linkType: 'STREAM',
+      statusHeader: 'This event has ended',
+      statusSubHeader: 'Couldn’t get a ticket? We got your back.',
+      link: undefined,
+    });
+  });
+
+  describe('getActionLink', () => {
+    let externalLinks;
+
+    beforeEach(() => {
```

Comment:
Why do we do this? Does out function mutate the input? If so we should fix that.